src/libmongoc/tests/test-libmongoc.c

```diff
@@ -1395,6 +1403,24 @@ test_framework_get_uri ()
    return uri;
 }
 
+mongoc_uri_t *
+test_framework_get_uri_multi_mongos_loadbalanced () {
+   char *uri_str_no_auth;
+   char *uri_str;
+   mongoc_uri_t *uri;
+
+   uri_str_no_auth = _mongoc_getenv ("MULTI_MONGOS_LB_URI");
+   if (!uri_str_no_auth) {
+      test_error ("expected MULTI_MONGOS_LB_URI to be set");
+   }
+   uri_str = test_framework_add_user_password_from_env (uri_str_no_auth);
+   uri = mongoc_uri_new (uri_str);
```

Comment:
Note that this can return NULL.-- which may be fine with us, as we wind up just returning it in any case.

Reply:
Thank you for the suggestion. Updating to `mongoc_uri_new_with_error` and `ASSERT_OR_PRINT` is a good idea. Done.